stdlib/asyncio/tasks.pyi

```diff
@@ -356,5 +359,40 @@ else:
 def current_task(loop: AbstractEventLoop | None = None) -> Task[Any] | None: ...
 def _enter_task(loop: AbstractEventLoop, task: Task[Any]) -> None: ...
 def _leave_task(loop: AbstractEventLoop, task: Task[Any]) -> None: ...
+
+if sys.version_info >= (3, 12):
+    _TaskT_co = TypeVar("_TaskT_co", bound=Task[Any], covariant=True)
+
+    class _CustomTaskConstructor(Protocol[_TaskT_co]):
+        def __call__(
+            self,
+            __coro: _TaskCompatibleCoro[Any],
+            loop: AbstractEventLoop,
+            name: str | None,
+            context: Context | None,
+            eager_start: bool,
+        ) -> _TaskT_co: ...
```

Comment:
I think the idea might be that it's permissible for the custom task constructor to return anything that _duck-types_ as an `asyncio.Task` -- it doesn't necessarily _have_ to be a subclass of `asyncio.Task`. That would be even more complicated than what we have currently, however... I'd vote for keeping it simple until we get a complaint that we're not allowing somebody's duck-type-compatible third-party `Task` implementation